File: src/main/java/com/itechart/demojavakotlin/service/MovieService.java
package com.itechart.demojavakotlin.service;

import com.itechart.demojavakotlin.entity.MovieEntity;
import com.itechart.demojavakotlin.model.MovieRequest;
import com.itechart.demojavakotlin.model.TicketsRequest;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public interface MovieService {

    List<MovieEntity> getAll();

    MovieEntity addMovie(final MovieRequest requestMovie);

    void deleteMovie(final UUID id);

    MovieEntity getMovieByName(final String movieName);

    void changeTotalTicketQuantity(final int quantity, final UUID id, final BigDecimal ticketPrice);

    void addTicketsToMovie(final TicketsRequest ticketsRequest);

    void removeTicketsFromMovie(final TicketsRequest ticketsRequest);
}

File: kotlinpath/demojavakotlin/src/main/kotlin/by/itechart/demojavakotlin/service/MovieService.kt
package by.itechart.demojavakotlin.service

import by.itechart.demojavakotlin.entity.MovieEntity
import by.itechart.demojavakotlin.model.MovieRequest
import by.itechart.demojavakotlin.model.TicketRequest
import java.math.BigDecimal
import java.util.*

interface MovieService {
    fun getAll(): List<MovieEntity>

    fun addMovie(requestMovie: MovieRequest): MovieEntity

    fun deleteMovie(id: UUID)

    fun getMovieByName(movieName: String): MovieEntity

    fun changeTotalTicketQuantity(quantity: Int, id: UUID, ticketPrice: BigDecimal)

    fun addTicketsToMovie(ticketRequest: TicketRequest)

    fun removeTicketsFromMovie(ticketRequest: TicketRequest)
}
